packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -103,6 +103,8 @@ class _ModalBottomSheetLayout extends OneChildLayoutDelegate {
     childTop.end = childSize.height;
     return new Point(0.0, size.height - childTop.value);
   }
+
+  bool shouldRelayout() => true;
```

Comment:
The BottomSheet layout is only built when the bottom sheet is animating, so there aren't any false positives.